### ISSUE TITLE
Add nixpkgs follows example

### ISSUE THREAD

**Issue by bburdette** [nix-community/naersk#333]
**TLDR:** would be nice to have an example of nixpkgs.follows usage since naersk seems to default to an old nixpkgs with an old cargo version.  

I've been working on a project lately that needs cargo --version to be 1.70 or greater.  

With `nix develop` my flake would show cargo of 1.83, but when I did `nix build` the naersk build would fail.  

Examining my flake.lock, it turns out a 2023 nixpkgs was being pulled in from someplace.  This is probably where the old cargo was coming from.  Commenting out naersk from the flake made that obsolete nixpkgs dependency go away.  

The solution was to add a `follows` to my flake inputs, like so:

```
  inputs = {
    nixpkgs.url = "github:NixOS/nixpkgs/nixos-24.05";
    flake-utils.url = "github:numtide/flake-utils";
    naersk.url = "github:nix-community/naersk";
    naersk.inputs.nixpkgs.follows = "nixpkgs";
  };
```

This seems like essential knowledge for folks using naersk to have, so it would be great to have an example added to the README to make it easier to figure out.
